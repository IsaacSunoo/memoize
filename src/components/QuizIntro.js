import React from 'react'

function QuizIntro(props) {
      return (
        <div className='quiz-intro-container'>
            <div className='quiz-intro'>
                <p>Test your knowledge on array prototypes.<br />
                    Each question provides a definition for a specific array prototype.<br />
                    Determine which prototype belongs to the definition<br />
                    Are you ready?</p>
            </div>
            <div className='start-quiz'>
                <button type='submit' onClick={props.display} className='start-quiz-btn'>Begin Quiz</button>
            </div>
        </div>
      )
}

export default QuizIntro;