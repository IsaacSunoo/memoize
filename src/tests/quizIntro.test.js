import React from 'react';
import QuizIntro from '../components/QuizIntro';
import { shallow } from 'enzyme';

const displayQuizMock = jest.fn();

describe('QuizIntro', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <QuizIntro
                display={displayQuizMock}/>
        )
    })

    it('should match the snapshot with all data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should invoke change display when clicked', () => {
        wrapper.find('.start-quiz-btn').simulate('click');
    })
});