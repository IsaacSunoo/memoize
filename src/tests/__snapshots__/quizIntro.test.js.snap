// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizIntro should match the snapshot with all data passed in correctly 1`] = `
<div
  className="quiz-intro-container"
>
  <div
    className="quiz-intro"
  >
    <p>
      Test your knowledge on array prototypes.
      <br />
      Each question provides a definition for a specific array prototype.
      <br />
      Determine which prototype belongs to the definition
      <br />
      Are you ready?
    </p>
  </div>
  <div
    className="start-quiz"
  >
    <button
      className="start-quiz-btn"
      onClick={[MockFunction]}
      type="submit"
    >
      Begin Quiz
    </button>
  </div>
</div>
`;
